main.go

```diff
@@ -165,21 +165,22 @@ func main() {
 			log.Info().Msg("Listing horizontal pod autoscalers for all namespaces...")
 			hpas, err := client.AutoscalingV1().ListHorizontalPodAutoscalers(context.Background(), k8s.AllNamespaces)
 			if err != nil {
-				log.Error().Err(err)
-			}
-			log.Info().Msgf("Cluster has %v horizontal pod autoscalers", len(hpas.Items))
-
-			// loop all hpas
-			if hpas != nil && hpas.Items != nil {
-				for _, hpa := range hpas.Items {
-					waitGroup.Add(1)
-					status, err := processHorizontalPodAutoscaler(client, hpa, "poller")
-					hpaTotals.With(prometheus.Labels{"namespace": *hpa.Metadata.Namespace, "status": status, "initiator": "poller"}).Inc()
-					waitGroup.Done()
-
-					if err != nil {
-						log.Error().Err(err)
-						continue
+				log.Error().Err(err).Msg("")
+			} else {
```

Comment:
Should we continue if there is an error getting the HPAs?
Otherwise we can just return inside the `if`.

Reply:
You mean return from the whole function, so exist from this application, effectively restarting the pod?  
Or just mean to have a `continue`, so that the `for` loop continues immediately without doing the wait?

I did it this way so that we do the wait before trying again.

I'm not sure which is the best approach out of these three though. @JorritSalverda, what was the original intention?

Reply:
Preferably on any error we don't exit the pod but wait and retry later. If the pod would run as a cronjob instead exiting would be fine, but since we handle the intervals inside this application better to just wait for the next time. When an error happens it might be interesting to track that in a separate prometheus timeline so we can alert on when we see more than x errors in a certain interval.

Reply:
@JorritSalverda Yeah, I was wondering what happens if one instance of the auto-scaler just keeps failing forever without ever succeeding.  
Maybe in a followup PR we can implement a counter that gets incremented on every consecutive failure, and it makes the process exit if it fails more than some `x` times in a row.